monai/utils/type_conversion.py

```diff
@@ -154,6 +155,49 @@ def convert_to_numpy(data, wrap_sequence: bool = False):
     return data
 
 
+def convert_to_cupy(data, dtype, wrap_sequence: bool = True):
+    """
+    Utility to convert the input data to a cupy array. If passing a dictionary, list or tuple,
+    recursively check every item and convert it to cupy array.
+
+    Args:
+        data: input data can be PyTorch Tensor, numpy array, cupy array, list, dictionary, int, float, bool, str, etc.
+            Tensor, numpy array, cupy array, float, int, bool are converted to cupy arrays
+
+            for dictionary, list or tuple, convert every item to a numpy array if applicable.
+        wrap_sequence: if `False`, then lists will recursively call this function. E.g., `[1, 2]` -> `[array(1), array(2)]`.
+            If `True`, then `[1, 2]` -> `array([1, 2])`.
+    """
+
+    # direct calls
+    if isinstance(data, cp_ndarray):
+        if dtype is not None:
```

Comment:
I think you don't need to check `None`.

Thanks.

Reply:
In cupy, `astype(None)` is equivalnet of `astype('float64')`, so I don't want to change the dtype if it is not intended to.

Reply:
Hi @drbeh ,

I checked the doc-string of `cupy.asarray()`, seems `None` means "It is inferred from the input by default":
https://docs.cupy.dev/en/stable/reference/generated/cupy.asarray.html

Thanks.